src/iosrv/ult.c

```diff
@@ -363,12 +363,35 @@ sched_ult2xs(int ult_type, int tgt_id)
 	switch (ult_type) {
 	case DSS_ULT_IOFW:
 	case DSS_ULT_MISC:
-		return (DSS_MAIN_XS_ID(tgt_id) + 1) % DSS_XS_NR_TOTAL;
+		if (!dss_helper_pool)
+			return (DSS_MAIN_XS_ID(tgt_id) + 1) % DSS_XS_NR_TOTAL;
+
+		if (dss_tgt_offload_xs_nr >= dss_tgt_nr)
+			return (dss_sys_xs_nr + dss_tgt_nr + tgt_id);
```

Comment:
if each VOS has its own private helper, the VOS XS and helper XS are close same as before (as above L367).

If all vos share a helper XS pool (for example 16 VOS use 6 helper XS), the current patch firstly create all the VOS xs and then all the helpers (as a shared pool). To avoid make it further complex.

Reply:
In this case "dss_tgt_offload_xs_nr >= dss_tgt_nr', each VOS XS will have it's own private helper, right?

Reply:
it is for the case for example 8 vos tgt, with 12 helpers. Then each vos has one private 1st helper, and all the 8 vos share the left 4 helpers as 2nd helper. 
I see what you mean that can make the IO XS and its 1st helper XS closer, this need some extra code change, can change it later when needed.